templates/prometheus.yaml

```diff
@@ -140,6 +140,22 @@ spec:
                   regex: '(.*):10250'
                   replacement: '${1}:1338'
                   target_label: __address__
+            - job_name: 'gpu_metrics'
+              scrape_interval: 1s
```

Comment:
Is there a reason we need a shorter scrape interval here? The other targets use the default `1m` interval; `1s` is relatively quite frequent.

Reply:
Will `1s` cause a burden on Prometheus? 

I just use the `scrape_interval` defined from the official Nvidia monitoring tools:
https://github.com/NVIDIA/gpu-monitoring-tools/blob/master/exporters/prometheus-dcgm/k8s/pod-gpu-metrics-exporter/prometheus/prometheus-configmap.yaml#L18

I can relax it if necessary. Or we can tune it in the future when there is a perf impact.

Reply:
I don't think a `1s` scrape interval will burden Prometheus. But there are a couple reasons to not set the interval here.

The first is that the scrape timeout can't exceed the scrape interval, and a `1s` timeout is pretty short. (The default timeout is `10s`.) So setting a short interval increases the likelihood that a scrape will be canceled if the target is slow to respond, leading to inconsistent collection and dropped metrics.

The other reason is that the Prometheus community recommends using the same scrape interval among different targets as much as possible. (Examples: [here](https://www.robustperception.io/keep-it-simple-scrape_interval-id), [here](https://github.com/prometheus/prometheus/issues/2353#issuecomment-273709709)) This makes configuration simpler, and also makes it easier to write queries. Queries that use functions like `rate()` need to take into account the sample rate of the underlying metric. This is easier to do if you can assume that all metrics are sampled at the same rate.

So if we don't have a compelling reason to set an interval here, let's use the default.

Reply:
Gotcha! Makes sense to me!

I will remove the 1s which would make it uses the default interval. Thanks, @branden !